scanner/src/main/kotlin/ScannerMatcher.kt

```diff
@@ -126,6 +101,65 @@ data class ScannerMatcher(
     }
 }
 
+/**
+ * A holder class for the [ScannerMatcher] configuration.
+ */
+data class ScannerMatcherConfig(
+    val regScannerName: String? = null,
+    val minVersion: String? = null,
+    val maxVersion: String? = null,
+    val configuration: String? = null
+) {
+    companion object {
+        val EMPTY = ScannerMatcherConfig(null, null, null, null)
+
+        /**
+         * The name of the property defining the regular expression for the scanner name as part of [ScannerMatcher].
+         * The [scanner details][ScannerDetails] of the corresponding scanner must match the criteria.
+         */
+        internal const val PROP_CRITERIA_NAME = "regScannerName"
+
+        /**
+         * The name of the property defining the minimum version of the scanner as part of [ScannerMatcher]. The
+         * [scanner details][ScannerDetails] of the corresponding scanner must match the criteria.
+         */
+        internal const val PROP_CRITERIA_MIN_VERSION = "minVersion"
+
+        /**
+         * The name of the property defining the maximum version of the scanner as part of [ScannerMatcher]. The
+         * [scanner details][ScannerDetails] of the corresponding scanner must match the criteria.
+         */
+        internal const val PROP_CRITERIA_MAX_VERSION = "maxVersion"
+
+        /**
+         * The name of the property defining the configuration of the scanner as part of [ScannerMatcher]. The
+         * [scanner details][ScannerDetails] of the corresponding scanner must match the criteria.
+         */
+        internal const val PROP_CRITERIA_CONFIGURATION = "configuration"
+
+        private val properties = listOf(
+            PROP_CRITERIA_NAME, PROP_CRITERIA_MIN_VERSION, PROP_CRITERIA_MAX_VERSION, PROP_CRITERIA_CONFIGURATION
+        )
+
+        /**
+         * Create a [ScannerMatcherConfig] from the provided options. Return the created config and the options without
+         * the properties that are used to configure the matcher.
+         */
+        fun create(options: Options): Pair<ScannerMatcherConfig, Options> {
+            val filteredOptions = options.filterKeys { it !in properties }
```

Comment:
Another thought: Now that we have a clean separation between `options` and `secrets`, it seems a bit weird to include *storage* related configuration into scanner options. It probably would be cleaner to introduce a third "top-level" `matcher` configuration map for these. Or even move the matcher options to the storage configuration (if [scan storages were plugins](https://github.com/oss-review-toolkit/ort/issues/6603)) as you might want to have more lenient matchers for storages that e.g. only offer older ScanCode results ("better take something than nothing"), but use more strict matchers for storages that *also* have newer ScanCode results.

Reply:
I'm also not super happy about the tight coupling of the matcher and the wrapper config, but I would prefer to keep it for now to not further complicate the changes. In a similar direction I was thinking that it could be useful to allow different configurations for the same scanner class, like we allow it for curation providers, for example to use different settings to scan projects and packages. However, since this and the feature you suggested to have different matchers for different storages are currently theoretical use cases without a real demand I think we should discuss these ideas first in the dev or community meetings.